dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/filter-fee-record-payment-entity-groups.test.ts

```diff
@@ -1,301 +1,285 @@
+        expect(result).not.toContainEqual(secondGroup);
       });
 
       it('returns all groups which have a facility id which partially matches the supplied facility id filter', () => {
         // Arrange
-        const anotherPayments = [PaymentEntityMockBuilder.forCurrency('GBP').withId(15).build()];
-        const anotherFeeRecords = [
-          FeeRecordEntityMockBuilder.forReport(aUtilisationReport())
-            .withId(feeRecordIdGenerator.next().value)
-            .withFacilityId('11112222') // partially matches firstFeeRecords[0].facilityId
-            .withPayments(anotherPayments)
-            .build(),
-        ];
-        const anotherFeeRecordPaymentGroup: FeeRecordPaymentEntityGroup = {
-          feeRecords: anotherFeeRecords,
-          payments: anotherPayments,
+        const firstGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('11111111'), aFeeRecordWithFacilityId('22222222')],
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
+        };
+
+        const secondGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('33333333')],
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
         };
 
+        const thirdGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('11112222')], // partially matches firstGroup.feeRecords[0].facilityId
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
+        };
+
+        const groups = [firstGroup, secondGroup, thirdGroup];
+
         const facilityIdFilter = '1111';
-        const paymentDetailsFilters = { facilityId: facilityIdFilter };
+        const filters = { facilityId: facilityIdFilter };
 
         // Act
-        const result = filterFeeRecordPaymentEntityGroupsByPaymentDetailsFilters(
-          [...allFeeRecordPaymentGroups, anotherFeeRecordPaymentGroup],
-          paymentDetailsFilters,
-        );
+        const result = filterFeeRecordPaymentEntityGroups(groups, filters);
 
         // Assert
-        expect(result).toContainEqual(firstFeeRecordPaymentGroup);
-        expect(result).not.toContainEqual(secondFeeRecordPaymentGroup);
-        expect(result).toContainEqual(anotherFeeRecordPaymentGroup);
+        expect(result).toContainEqual(firstGroup);
+        expect(result).not.toContainEqual(secondGroup);
+        expect(result).toContainEqual(thirdGroup);
       });
     });
 
     describe('payment currency filter', () => {
       it('returns the group where the supplied payment currency filter matches one of the payments in a group', () => {
         // Arrange
-        const paymentCurrencyFilter = secondPayments[0].currency;
-        const paymentDetailsFilters = { paymentCurrency: paymentCurrencyFilter };
+        const firstGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('11111111')],
```

Comment:
Given the facility id is irrelevant could we just use `FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).build()`? I think it makes it clearer that it isn't relevant.